src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/SchemaFormStore.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import {computed} from 'mobx';
+import type {FormStoreInterface, RawSchema, Schema} from '../types';
+import metadataStore from './metadataStore';
+
+export default class SchemaFormStore implements FormStoreInterface {
+    innerFormStore: ?FormStoreInterface;
+
+    constructor(formKey: string, initializer: (schema: RawSchema, jsonSchema: Object) => FormStoreInterface) {
```

Comment:
I think we should move the `initializer` first, because then it is easier to add more parameters like `formKey` at the end of the function signature.

Reply:
Not sure if it is a good idea, but we could even make a `Promise<FormStoreInterface>` the only argument of the constructor and call it something like `DeferredFormStore`. Both ways, i feel like its hard to find a good name for this.

Reply:
Don't know, that feels somehow weird to me :see_no_evil: And I would have to create another `FormStore` implementation... Also, I don't think that there'll be huge BC issues here, because people should use the factory.